internal/services/azapi_resource.go

```diff
@@ -929,6 +938,40 @@ func expandBody(body map[string]interface{}, model AzapiResourceModel) diag.Diag
 	return diag.Diagnostics{}
 }
 
+func preflightValidation(ctx context.Context, client *clients.ResourceClient, requestPlan *AzapiResourceModel, body map[string]interface{}) error {
+	azureResourceType, apiVersion, err := utils.GetAzureResourceTypeApiVersion(requestPlan.Type.ValueString())
+	if err != nil {
+		return err
+	}
+
+	if !utils.IsTopLevelResourceType(azureResourceType) {
+		return nil
+	}
+
+	resourceBody := make(map[string]interface{})
+	for k, v := range body {
+		resourceBody[k] = v
+	}
+
+	resourceBody["name"] = requestPlan.Name.ValueString()
+	resourceBody["apiVersion"] = apiVersion
+	requestBody := make(map[string]interface{})
+	requestBody["provider"], requestBody["type"], _ = strings.Cut(azureResourceType, "/")
+	if locationValue, ok := resourceBody["location"]; ok {
+		requestBody["location"] = locationValue
+	}
+
+	requestBody["scope"] = requestPlan.ParentID.ValueString()
+	requestBody["resources"] = []map[string]interface{}{resourceBody}
+
+	_, err = client.Action(ctx, "/providers/Microsoft.Resources", "validateResources", "2020-10-01", "POST", requestBody)
```

Comment:
Please define the request body struct for validateResources API instead of using the generic struct.

Reply:
updated. Thanks.